### PR TITLE
Fix redirects

### PR DESCRIPTION
**Pull request by hvelarde**

### CODE REVIEW

**Comment by hvelarde** [comment]
I don't understand why the build fails; I think the linter is returning a wrong code as WARNINGs should not be interpreted as failures.

**Comment by gforcada** [comment]
@hvelarde yes, a warning would be best, but so far I see 2 reported errors:

```
(line   29) broken    https://www.dipf.de/en/research/projects - <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)>
```

and 

```
(line  228) broken    http://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html/download.html - HTTP Error 404: Not Found
```

**Comment by hvelarde** [comment]
that's totally broken: the certificate of **www.dipf.de** is valid and there are no references to something like `http://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html/`download.html in the whole repository.

certificate viewer www dipf de _023

**Comment by gforcada** [comment]
@hvelarde can you please refrain from this tone? It's discouraging to help someone that replies like that...

As for the link that does not exist... well, it does, just that as there is probably a redirect the URL is slightly different. Search for the domain and you will find it: https://github.com/plone/training/search?utf8=%E2%9C%93&q=chiark.greenend.org&type=

https://github.com/plone/training/blob/2e3ac217edde40a8b7d138e68a83915eb4b1c5a5/plone_training_config/instructions.rst

As for the certificate, that could easily be curl/wget/python that have an older version (I remember some problems with python versions earlier than 2.7.12 or something around that version.

File: _locales/anatomy.pot [hunk 1]
```diff
@@ -188,12 +188,12 @@ msgstr ""
 
 #: ../anatomy.rst:92
 # 02091034744649798c63715649bc667f
-msgid "`Pyramid <http://docs.pylonsproject.org/en/latest/docs/pyramid.html>`_ is a Python web application development framework that is often seen as the successor to Zope."
+msgid "`Pyramid <https://trypyramid.com/resources.html>`_ is a Python web application development framework that is often seen as the successor to Zope."
```
Comment (stevepiercy) [inline _locales/anatomy.pot]: s/b just https://trypyramid.com

File: _locales/anatomy.pot [hunk 2]
```diff
@@ -188,12 +188,12 @@ msgstr ""
...
 msgstr ""
 
 #: ../anatomy.rst:93
 # e582e67056f14b83b5b193ff2bc9c942
-msgid "It does less than Zope, is very pluggable and `uses the Zope Component Architecture <http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_."
+msgid "It does less than Zope, is very pluggable and `uses the Zope Component Architecture <https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_."
```
Comment (stevepiercy) [inline _locales/anatomy.pot]: The link here is for Pyramid docs that use ZCA, not for ZCA itself, which is:
https://docs.zope.org/zope.component/narr.html

File: _locales/es/LC_MESSAGES/anatomy.po [hunk 3]
```diff
@@ -261,18 +261,18 @@ msgstr "Pyramid"
 # 02091034744649798c63715649bc667f
 #: ../anatomy.rst:92
 msgid ""
-"`Pyramid <http://docs.pylonsproject.org/en/latest/docs/pyramid.html>`_ is a "
+"`Pyramid <https://trypyramid.com/resources.html>`_ is a "
 "Python web application development framework that is often seen as the "
```
Comment (stevepiercy) [inline _locales/es/LC_MESSAGES/anatomy.po]: ibid, https://trypyramid.com please

File: _locales/es/LC_MESSAGES/anatomy.po [hunk 4]
```diff
@@ -261,18 +261,18 @@ msgstr "Pyramid"
...
 "successor to Zope."
-msgstr "`Pyramid <http://docs.pylonsproject.org/en/latest/docs/pyramid.html>`_ es un framework de desarrollo de aplicaciones Web en Python que es a veces visto como el sucesor natural de Zope."
+msgstr "`Pyramid <https://trypyramid.com/resources.html>`_ es un framework de desarrollo de aplicaciones Web en Python que es a veces visto como el sucesor natural de Zope."
```
Comment (stevepiercy) [inline _locales/es/LC_MESSAGES/anatomy.po]: ibid, https://trypyramid.com please

File: _locales/es/LC_MESSAGES/anatomy.po [hunk 5]
```diff
@@ -261,18 +261,18 @@ msgstr "Pyramid"
...
 
 # e582e67056f14b83b5b193ff2bc9c942
 #: ../anatomy.rst:93
 msgid ""
 "It does less than Zope, is very pluggable and `uses the Zope Component "
 "Architecture "
-"<http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_."
-msgstr "Eso hace menos que Zope, es muy configurable y `usa la Arquitectura de Componentes Zope <http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_."
+"<https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_."
```
Comment (stevepiercy) [inline _locales/es/LC_MESSAGES/anatomy.po]: ibid, https://docs.zope.org/zope.component/narr.html

File: _locales/es/LC_MESSAGES/anatomy.po [hunk 6]
```diff
@@ -261,18 +261,18 @@ msgstr "Pyramid"
...
+msgstr "Eso hace menos que Zope, es muy configurable y `usa la Arquitectura de Componentes Zope <https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_."
```
Comment (stevepiercy) [inline _locales/es/LC_MESSAGES/anatomy.po]: https://docs.zope.org/zope.component/narr.html

File: mastering_plone/anatomy.rst [hunk 7]
```diff
@@ -120,8 +120,8 @@ The `Zope Component Architecture <http://muthukadan.net/docs/zca.html>`_, which
 Pyramid
 -------
 
-* `Pyramid <http://docs.pylonsproject.org/en/latest/docs/pyramid.html>`_ is a Python web application development framework that is often seen as the successor to Zope.
-* It does less than Zope, is very pluggable and `uses the Zope Component Architecture <http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_ “under the hood” to perform view dispatching and other application configuration tasks.
+* `Pyramid <https://trypyramid.com/resources.html>`_ is a Python web application development framework that is often seen as the successor to Zope.
```
Comment (stevepiercy) [inline mastering_plone/anatomy.rst]: ibid, https://trypyramid.com please

File: mastering_plone/anatomy.rst [hunk 8]
```diff
@@ -120,8 +120,8 @@ The `Zope Component Architecture <http://muthukadan.net/docs/zca.html>`_, which
...
+* It does less than Zope, is very pluggable and `uses the Zope Component Architecture <https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/zca.html>`_ “under the hood” to perform view dispatching and other application configuration tasks.
```
Comment (stevepiercy) [inline mastering_plone/anatomy.rst]: https://docs.zope.org/zope.component/narr.html

**Comment by stevepiercy** [comment]
Out of curiosity, how did y'all run a link checker? I have used Sphinx's `make linkcheck` which provides a log that indicates both original and their redirected URLs. That helped me find dozens of outdated redirects for http to https, readthedocs.org to .io, and others.

**Comment by hvelarde** [comment]
@gforcada sorry to hear that but IMO I'm not using any aggressive tone neither I'm attacking you; I'm just been pretty direct to the point saying I found those tests results broken because of the reasons I stated above. now you're providing more feedback that could be useful but I don't know how to fix that.

@stevepiercy I just replaced the links with what the linter reported; seems those of Pyramid have to be fixed there also.

**Comment by hvelarde** [comment]
gentlemen, the purpose of this PR is to fix the redirects; if you agree now they are fine, please merge.

@gforcada you can fix those errors in another PR; the build has been failing for some time as you can see in the [build history](https://travis-ci.org/plone/training/builds).

**Comment by hvelarde** [comment]
@pbauer there are other links pointing there, please review 9c3cb5f9f4204cfdc001454e3cd1efc64f9ed984 and talk to @stevepiercy as he seems to have a different idea.

**Comment by stevepiercy** [comment]
@pbauer please see [my comment regarding ZCA](https://github.com/plone/training/pull/266/files/4a40b799f50011b91eaf8049ec6efd7baf3bff33#r124483385). I assume it is better to link to the actual ZCA docs, not Pyramid's docs about ZCA.

**Comment by stevepiercy** [comment]
@gforcada merged this before the discussion was resolved. If someone else agrees that we should use the canonical link to ZCA instead of Pyramid docs, I can open a new PR for it. Please let me know.

**Comment by gforcada** [comment]
@stevepiercy open a new pr so it can be discussed 👍

**Comment by pbauer** [comment]
I fixed one broken link and added the one where it complained about to the ssl-cert to the ingores-list: https://github.com/plone/training/commit/5a52e1abb01ec15685b98c73d96b0611066aa3ef

The auto-deployment worked fine now and the changes are already online.

**Pull request merged**